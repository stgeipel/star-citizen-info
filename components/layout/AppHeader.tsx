import * as React from 'react';
import clsx from 'clsx';
import Link from 'next/link';
import dynamic from 'next/dynamic';

import { FiMoreHorizontal } from 'react-icons/fi';
import { AiOutlineFontColors } from 'react-icons/ai';
import { MdFeaturedPlayList } from 'react-icons/md';

const ThemeSwitch = dynamic(() => import('@components/layout/ThemeSwitch'), {
  ssr: false,
});

export interface Props {
  className?: string | Record<string, unknown>;
}

const AppHeader: React.FC<Props> = ({ className }) => {
  return (
    <header
      className={clsx(
        'bg-white dark:bg-gray-800 py-2 px-3 flex items-center sticky top-0',
        className
      )}
    >
      <nav className="flex items-center justify-between flex-wrap bg-background-primary p-4 shadow-xl">
        <div className="flex items-center flex-shrink-0  mr-6">
          <span className="font-semibold text-xl tracking-tight text-copy-primary">
            Star Citizen Info
          </span>

          {/* <div className="hidden md:ml-6 md:flex md:items-center">
        <a
          href="#"
          className="px-3 py-2 rounded-md text-sm font-medium leading-5 text-white bg-gray-900 focus:outline-none focus:text-white focus:bg-gray-700 transition duration-150 ease-in-out"
        >
          Dashboard
        </a>
        <a
          href="#"
          className="ml-4 px-3 py-2 rounded-md text-sm font-medium leading-5 text-gray-300 hover:text-white hover:bg-gray-700 focus:outline-none focus:text-white focus:bg-gray-700 transition duration-150 ease-in-out"
        >
          Team
        </a>
        <a
          href="#"
          className="ml-4 px-3 py-2 rounded-md text-sm font-medium leading-5 text-gray-300 hover:text-white hover:bg-gray-700 focus:outline-none focus:text-white focus:bg-gray-700 transition duration-150 ease-in-out"
        >
          Projects
        </a>
        <a
          href="#"
          className="ml-4 px-3 py-2 rounded-md text-sm font-medium leading-5 text-gray-300 hover:text-white hover:bg-gray-700 focus:outline-none focus:text-white focus:bg-gray-700 transition duration-150 ease-in-out"
        >
          Calendar
        </a>
      </div> */}
        </div>
        {/* <!-- right side of nav --> */}
        <div className="absolute inset-y-0 right-0 flex items-center pr-2 sm:static sm:inset-auto sm:ml-6 sm:pr-0">
          <div className="ml-3 relative">
            <div>
              <div className="flex items-center justify-center">
                <label
                  htmlFor="toogleA"
                  className="flex items-center cursor-pointer"
                >
                  <div className="relative">
                    <input id="toogleA" type="checkbox" className="hidden" />
                    <div className="toggle__line w-10 h-4 bg-gray-400 rounded-full shadow-inner"></div>
                    <div className="toggle__dot absolute w-6 h-6 bg-white rounded-full shadow inset-y-0 left-0"></div>
                  </div>
                  <div className="ml-3 text-gray-700 font-medium"></div>
                </label>
              </div>
            </div>
          </div>
        </div>
      </nav>
    </header>
  );
};

export default AppHeader;
