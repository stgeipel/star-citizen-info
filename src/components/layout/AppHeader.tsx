import * as React from 'react';
import clsx from 'clsx';
import Link from 'next/link';
import dynamic from 'next/dynamic';
import { FiMoreHorizontal } from 'react-icons/fi';
import { FaPrayingHands } from 'react-icons/fa';
import Dropdown from '../Dropdown';
import routes from '@utils/routes';

const ThemeSwitch = dynamic(() => import('@components/layout/ThemeSwitch'), {
  ssr: false,
});

export interface Props {
  className?: string | Record<string, unknown>;
}

const AppHeader: React.FC<Props> = ({ className }) => {
  return (
    <header
      className={clsx(
        'bg-white dark:bg-gray-800 py-2 px-3 flex items-center sticky top-0',
        className
      )}
    >
      <div className="mr-auto flex items-center">
        <Link {...routes.INDEX}>
          <a
            title="home"
            className="flex-grow text-xl font-black text-black dark:text-white dark:hover:text-black"
          >
            Star Citizen Info
          </a>
        </Link>
      </div>
      <div>
        <Dropdown buttonLabel={<FiMoreHorizontal />}>
          <ul className="w-40 space-y-4 py-2 text-sm">
            <li>
              <Link {...routes.THANKSTO}>
                <a className="text-primary hover:underline" title="Typography">
                  <FaPrayingHands className="inline mr-2" />
                  Thanks To
                </a>
              </Link>
            </li>
            <li>
              <hr className="dark:border-gray-700" />
            </li>
            <li className="flex items-center justify-between">
              <span className="mr-auto text-gray-600">Theme</span>
              <ThemeSwitch />
            </li>
          </ul>
        </Dropdown>
      </div>
    </header>
  );
};
export default AppHeader;
