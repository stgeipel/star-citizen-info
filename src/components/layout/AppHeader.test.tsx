import * as React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import AppHeader from './AppHeader';

describe('AppHeader', () => {
  it('matches snapshot', () => {
    const wrapper = shallow(<AppHeader />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
