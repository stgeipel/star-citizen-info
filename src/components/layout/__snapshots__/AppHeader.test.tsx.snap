// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppHeader matches snapshot 1`] = `
<header
  className="bg-white dark:bg-gray-800 py-2 px-3 flex items-center sticky top-0"
>
  <div
    className="mr-auto flex items-center"
  >
    <Link
      href="#"
    >
      <a
        className="flex-grow text-xl font-black text-black dark:text-white dark:hover:text-black"
        title="home"
      >
        Star Citizen Info
      </a>
    </Link>
  </div>
  <div>
    <Dropdown
      buttonLabel={<FiMoreHorizontal />}
    >
      <ul
        className="w-40 space-y-4 py-2 text-sm"
      >
        <li
          className="flex items-center justify-between"
        >
          <span
            className="mr-auto text-gray-600"
          >
            Theme
          </span>
          <ThemeSwitch />
        </li>
      </ul>
    </Dropdown>
  </div>
</header>
`;
